sentry-android-okhttp/src/main/java/io/sentry/android/okhttp/SentryOkHttpInterceptor.kt

```diff
@@ -104,6 +124,115 @@ class SentryOkHttpInterceptor(
+        // url example: https://api.github.com/users/getsentry/repos/#fragment?query=query
+        // url will be: https://api.github.com/users/getsentry/repos/
+        // ideally we'd like a parameterized url: https://api.github.com/users/{user}/repos/
+        // but that's not possible
+        var requestUrl = request.url.toString()
+
+        val query = request.url.query
+        if (!query.isNullOrEmpty()) {
+            requestUrl = requestUrl.replace("?$query", "")
+        }
+
+        val urlFragment = request.url.fragment
+        if (!urlFragment.isNullOrEmpty()) {
+            requestUrl = requestUrl.replace("#$urlFragment", "")
+        }
+
+        if (!captureFailedRequests ||
+            !PropagationTargetsUtils.contain(failedRequestTargets, requestUrl) ||
+            !containsStatusCode(response.code)
+        ) {
+            return
+        }
+
+        val mechanism = Mechanism().apply {
+            type = "SentryOkHttpInterceptor"
+        }
+        val exception = SentryHttpClientException(
+            "Event was captured because the request status code was ${response.code}"
+        )
+        val mechanismException = ExceptionMechanismException(mechanism, exception, Thread.currentThread(), true)
+        val event = SentryEvent(mechanismException)
+
+        val hint = Hint()
+        hint.set("request", request)
+        hint.set("response", response)
+
+        // remove after fields indexed
+//        val tags = mutableMapOf<String, String>()
+//        tags["status_code"] = response.code.toString()
+//        tags["url"] = requestUrl
+
+        val sentryRequest = io.sentry.protocol.Request().apply {
+            url = requestUrl
+            // Cookie is only sent if isSendDefaultPii is enabled
+            cookies = if (hub.options.isSendDefaultPii) request.headers["Cookie"] else null
+            method = request.method
+            queryString = query
+            headers = getHeaders(request.headers)
+            fragment = urlFragment
+
+            request.body?.contentLength().ifHasValidLength {
+                // should be mapped in relay and added to the protocol, right now
+                // relay isn't retaining unmapped fields
+                bodySize = it
+            }
+        }
+
+        val sentryResponse = io.sentry.protocol.Response().apply {
+            // Cookie is only sent if isSendDefaultPii is enabled due to PII
+            cookies = if (hub.options.isSendDefaultPii) response.headers["Cookie"] else null
+            headers = getHeaders(response.headers)
+            statusCode = response.code
+
+            response.body?.contentLength().ifHasValidLength {
+                bodySize = it
+            }
+        }
+
+//        event.tags = tags
+        event.request = sentryRequest
+        event.contexts.setResponse(sentryResponse)
+
+        hub.captureEvent(event, hint)
+    }
+
+    private fun containsStatusCode(statusCode: Int): Boolean {
```

Comment:
Maybe this could be called something like `shouldStatusCodeCauseSentryEvent` or similar?

Also should it be extracted into some util so it can be reused from other integrations?

Reply:
I don't think so, `isInRange` already can be reused.
In case the language has a built-in type for Range instead of `HttpStatusCodeRange`, they will need to implement this anyway.
Java does not have Range obj as part of the language.